### PR TITLE
Deprecate original_foreman_spec

### PR DESCRIPTION
**Pull request by mdellweg**
I had the feeling, there were two concepts competing for the same task.

### CODE REVIEW

**Comment by evgeni** [comment]
any reason `foreman_job_template.py` is executable now?

**Comment by evgeni** [comment]
looks correct to me, but I'll leave it to the designated reviewer ;)

**Comment by mdellweg** [comment]
ping @bagasse

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -366,22 +363,14 @@ def list_resource(self, resource, search=None, params=None):
...
-    def find_resource(self, resource, search, name=None, params=None, failsafe=False, thin=None):
+    def find_resource(self, resource, search, params=None, failsafe=False, thin=None):
         list_params = {}
         if params is not None:
             list_params.update(params)
         if thin is None:
             thin = self._thin_default
         list_params['thin'] = thin
         results = self.list_resource(resource, search, list_params)
-        if resource == 'snapshots':
```
Comment (evgeni) [inline plugins/module_utils/foreman_helper.py]: this is not needed anymore?

Comment (mdellweg) [reply plugins/module_utils/foreman_helper.py]: Nope, it has been "fixed" on the server side.

**Comment by bagasse** [comment]
@mdellweg, sorry for the delay, and thanks a lot for the refactor/cleanup !

**Pull request merged**